doc/MGMT_CLI.md

```diff
@@ -0,0 +1,420 @@
+
+No new containers are introduced for this feature. Existing Mgmt containers are updated.
+
+### 1.3.3 SAI Overview
+
+No new or existing SAI services are required
+
+
+# 2 Functionality
+
+* A sonic model will be created to map to existing redis DB keys for the fields to allow for backwards compatibility. 
+* A CLI XML template with an actioner script will be created to configure the appropriate model fields.
+
+
+# 3 Design
+## 3.1 Overview
+
+This feature will involve the mgmt-framework and telemetry containers as well as the host-services on the host. It will be used to modify configuration held in the config DB in redis.
+
+### 3.1.1 Service and Docker Management
+
+No new dockers will be introduced. Host services will be used to restart and/or stop/start the REST and telemetry services running in their respective containers.
+
+### 3.1.2 Packet Handling
+N/A
+
+## 3.2 DB Changes
+All of the fields except "enable" already exist in the CONFIG DB. However, they were not mapped to a sonic yang, and the format of some of the fields is changed. This requires a DB migration script. 
+
+  * PORT - <uint16> 1-65535 : Default 443 for REST, 8080 for Telemetry
+  * jwt_valid - <uint64> 1-2^64-1 : Default: 3600 (Seconds)
+  * jwt_refresh - <uint64> 1-2^64-1 : Default: 900 (Seconds)
+  * enable - <boolean> - Default: true
+  * client_auth - <enum> - Default: PASSWORD_JWT
+
+
+### 3.2.1 CONFIG DB
+
+  * REST_SERVER|default port - Already exists and will not change
+  * REST_SERVER|default client_auth - Already exists, but format is changing from string with comma separated values to enum of valid client auth modes.
+  * REST_SERVER|default jwt_valid - Already exists and will not change.
+  * REST_SERVER|default jwt_refresh - Already exists and will not change.
+  * REST_SERVER|default enable - Needs to be added, will be a "bool" type.
+
+  * TELEMETRY|gnmi port - Already exists and will not change
+  * TELEMETRY|gnmi client_auth - Already exists, but format is changing from string with comma separated values to enum of valid client auth modes.
+  * TELEMETRY|gnmi jwt_valid - Already exists and will not change.
+  * TELEMETRY|gnmi jwt_refresh - Already exists and will not change.
+  * TELEMETRY|gnmi enable - Needs to be added, will be a "bool" type.
+
+## 3.3 Switch State Service Design
+N/A
+
+### 3.6.2 CLI
+
+This feature uses the KLISH CLI only.
+  
+
+#### 3.6.2.1 Configuration Commands
+sonic(config)# management
+  rest       Authentication for REST
+  telemetry  Authentication for telemetry
+ 
+sonic(config)# management rest
+  client-authentication 
+  enable (default)
+  port
+  token-refresh          Configure the seconds before JWT expiry that the token can be refreshed
+  token-validity         Configure the seconds JWT token is valid for
+ 
+sonic(config)# management rest client-authentication
+  certificate               Certificate-based authentication only
+  jwt-certificate           Certificate and token-based authentication
```

Comment:
Pre-defined multi word token approach is not extensible. We will have to add many more tokens if we support a new mode in future (like oauth2). This will be confusing to the user.

Can we accept modes as separate tokens? Like:

```
management rest client-authentication jwt certificate
```